test/crc32_test.c

```diff
@@ -0,0 +1,201 @@
+/* crc32_test.c -- crc32 unit test
+ * Copyright (C) 2019-2021 IBM Corporation
+ * Authors: Rogerio Alves    <rogealve@br.ibm.com>
+ *          Matheus Castanho <msc@linux.ibm.com>
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+typedef unsigned long crc32_type;
```

Comment:
I'm not sure this is entirely correct. It should be `typedef z_crc_t crc32_type` correct? Because depending on the version of zlib and config it can be either unsigned int or unsigned long. Some older versions of zlib don't have `z_crc_t` either.

Reply:
Maybe for this file it is better to just use `unsigned long` every where and it will be cast to `uint32_t` in `crc32_z` call..

Reply:
Fair enough. I changed the code to use `unsigned long` everywhere.